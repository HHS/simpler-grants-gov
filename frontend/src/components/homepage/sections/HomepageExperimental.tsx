import { UswdsIconNames } from "src/types/generalTypes";

import { useMessages, useTranslations } from "next-intl";
import Link from "next/link";
import { Button, Grid } from "@trussworks/react-uswds";

import HomePageSection from "src/components/homepage/homePageSection";
import IconInfo from "src/components/homepage/IconInfoSection";

const ExperimentalContent = () => {
  const t = useTranslations("Homepage.sections.experimental");
  const messages = useMessages() as unknown as IntlMessages;
  const { iconSections } = messages.Homepage.sections.experimental;

  return (
    <HomePageSection className="bg-base-lightest" title={t("title")}>
      <h3 data-testid="homepage-experimental">{t("canDoHeader")}</h3>
      <h4>{t("canDoSubHeader")}</h4>
      <p>{t("canDoParagraph")}</p>
      <Link href="/search">
        <Button
          className="margin-y-2 usa-button--secondary"
          type="button"
          size="big"
        >
          {t("tryLink")}
        </Button>
      </Link>
      <h4>{t("cantDoHeader")}</h4>
      <p>{t("cantDoParagraph")}</p>
      <Grid row className="padding-y-2" gap="md">
        {iconSections.map((iconSection, iconSectionIdx) => (
          <Grid col={6} key={`experimental-iconsection-${iconSectionIdx}`}>
            <IconInfo
              description={iconSection.description}
              iconName={iconSection.iconName as UswdsIconNames}
              link={iconSection.http}
              linkText={iconSection.link}
              title={iconSection.title}
            />
          </Grid>
        ))}
      </Grid>
    </HomePageSection>
  );
};

export default ExperimentalContent;
