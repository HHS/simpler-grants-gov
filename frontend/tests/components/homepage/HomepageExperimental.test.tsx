import { render, screen } from "tests/react-utils";

import HomepageExperimental from "src/components/homepage/sections/HomepageExperimental";

describe("Homepage Experimental Content", () => {
  it("matches snapshot", () => {
    const { container } = render(<HomepageExperimental />);
    expect(container).toMatchSnapshot();
  });
  it("Renders without errors", () => {
    render(<HomepageExperimental />);
    const component = screen.getByTestId("homepage-experimental");

    expect(component).toBeInTheDocument();
  });
});
