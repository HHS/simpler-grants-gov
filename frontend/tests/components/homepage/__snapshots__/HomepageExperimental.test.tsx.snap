// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Experimental Content matches snapshot 1`] = `
<div>
  <div
    class="bg-base-lightest"
  >
    <div
      class="grid-container display-flex flex-column padding-y-4 grid-container tablet-lg:flex-row tablet-lg:padding-y-6"
      data-testid="gridContainer"
    >
      <div
        class="grid-row"
        data-testid="grid"
      >
        <div
          class="tablet-lg:grid-col-4"
          data-testid="grid"
        >
          <h2
            class="margin-0 font-sans-lg tablet:font-sans-xl tablet-lg:maxw-card-lg"
          >
            Test out our experimental features
          </h2>
        </div>
        <div
          class="tablet-lg:grid-col-8 margin-top-2 margin-bottom-0 tablet-lg:margin-0"
          data-testid="grid"
        >
          <h2
            data-testid="homepage-experimental"
          >
            Tell us what's working (and what's not)
          </h2>
          <h3>
            What you can do now
          </h3>
          <p
            class="font-sans-md line-height-sans-4"
          >
            Search real Grants.gov data here on Simpler.Grants.gov. Our search aims to deliver closer matches to your keywords and filters. We also redesigned the results and opportunity listings to make them easier to navigate and read.
          </p>
          <a
            href="/search"
          >
            <button
              class="usa-button usa-button--big margin-y-2 usa-button--secondary"
              data-testid="button"
              type="button"
            >
              Try the new simpler search
            </button>
          </a>
          <h3>
            What you can't do quite yet
          </h3>
          <p
            class="font-sans-md line-height-sans-4"
          >
            For now, you need to visit Grants.gov to access more advanced features like applying. We're working to bring the application process to this website soon. In the meantime, follow our roadmap to stay updated on our progress.
          </p>
          <div
            class="grid-row grid-gap-md padding-y-2"
            data-testid="grid"
          >
            <div
              class="grid-col-6"
              data-testid="grid"
            >
              <svg
                aria-hidden="true"
                class="usa-icon usa-icon--size-4 text-middle"
                role="img"
              >
                <use
                  href="/img.jpg#build"
                />
              </svg>
              <h3>
                Send us your feedback and suggestions.
              </h3>
              <p
                class="font-sans-md line-height-sans-4"
              >
                We're excited to hear from you to learn how we can improve.
              </p>
              <a
                class="font-sans-md line-height-sans-4"
                href="mailto:simpler@grants.gov"
              >
                Contact us at simpler@grants.gov
              </a>
            </div>
            <div
              class="grid-col-6"
              data-testid="grid"
            >
              <svg
                aria-hidden="true"
                class="usa-icon usa-icon--size-4 text-middle"
                role="img"
              >
                <use
                  href="/img.jpg#mail"
                />
              </svg>
              <h3>
                Be the first to hear about new features.
              </h3>
              <p
                class="font-sans-md line-height-sans-4"
              >
                Our newsletter delivers the latest news straight to your inbox.
              </p>
              <a
                class="font-sans-md line-height-sans-4"
                href="/subscribe"
              >
                Subscribe to our newsletter
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
